### PR TITLE
TIG-3391 Upgrade C++ Driver for Genny

### PR DESCRIPTION
**Pull request by aggrand**
I think we can expect RHEL 7 build to fail due to to the errors on the toolchain, but that's unrelated to our recent work and shouldn't affect production anyway because of [TIG-2860](https://jira.mongodb.org/browse/TIG-2860). I think we should go ahead with the upgrade because [TIG-3753](https://jira.mongodb.org/browse/TIG-3753) is affecting production anyway and that fix would be lumped into this upgrade. 

If this is agreed to merge, I can cut a followup ticket to either deal with the RHEL7 issue or just stop building the toolchain on it. Based on a conversation with @richardsamuels  it seems to be a rhel licensing issue which we'll either have to file a dependent build ticket for, or else find a hacky workaround.

### CODE REVIEW

**Comment by aggrand** [comment]
Patch sanity-checking an affected task: https://spruce.mongodb.com/version/61e864122fbabe701b13a434/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC

**Comment by aggrand** [comment]
Also a patch running microbenchmarks to make sure we're fine without rhel7 (the historic use of it, nowadays seems to only be rhel6)

**Review by vrachev — APPROVED** [review]
rhel7 not working should break m60 I think ([based on this](https://github.com/10gen/dsi/blob/41f94803c9287ec124002375cf29c56c45565ff7/configurations/workload_setup/workload_setup.common.yml#L44-L45)). I'm ok with fixing that in a later ticket since we need to unblock genny users now.

**Review by richardsamuels — APPROVED** [review]

**Comment by aggrand** [comment]
Another patch after some updates: https://spruce.mongodb.com/version/61faf40757e85a0bcfa14097/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC

File: src/lamplib/src/genny/toolchain.py [hunk 1]
```diff
@@ -157,12 +157,12 @@ class ToolchainDownloader(Downloader):
     # These build IDs are from the genny-toolchain Evergreen task.
     # https://evergreen.mongodb.com/waterfall/genny-toolchain
     # Find a compile task (for any build variant) and modify the URL:
-    # genny_toolchain_amazon2_t_compile_cf1c50fab5291cd583a5036dc9ade265bca165a3_21_10_20_20_05_26
-    # =>                                cf1c50fab5291cd583a5036dc9ade265bca165a3_21_10_20_20_05_26
+    # genny_toolchain_archlinux_t_compile_82eb7c32ad09726f3ef0ddc8d7f24a18b03d9644_21_11_23_16_37_21
+    # =>                                  82eb7c32ad09726f3ef0ddc8d7f24a18b03d9644_21_11_23_16_37_21
```
Comment (richardsamuels) [inline src/lamplib/src/genny/toolchain.py:161]: Random question, but why did this change to something else? Are we keeping this up to date with `TOOLCHAIN_BUILD_ID`?

**Review by richardsamuels — APPROVED** [review]
LGTM aside from my comment

File: src/lamplib/src/genny/toolchain.py (hunk 1 above)
Comment (aggrand) [reply src/lamplib/src/genny/toolchain.py:161]: It changed during the unrevert. I didn't think we were keeping it up-to-date, but I can do that.

**Comment by aggrand** [comment]
evergreen merge

**Pull request merged**